test/testhelper/mock_registry.go

```diff
@@ -0,0 +1,59 @@
+package testhelper
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+)
+
+type RegistryCtrl struct {
+	base string
+}
+
+func MockRegistry(base string) *httptest.Server {
+	rctrl := &RegistryCtrl{base: base}
+
+	handler := http.NewServeMux()
+	handler.HandleFunc("/v2/", rctrl.HandleV2)
+	return httptest.NewServer(handler)
+}
+
+func (rctrl RegistryCtrl) HandleV2(w http.ResponseWriter, r *http.Request) {
+	// log.Printf("Received request: %s %s", r.Method, r.URL.Path)
+
+	path := r.URL.Path[len("/v2/"):]
+
+	if path == "" {
+		ok(w)
+	} else {
+		rctrl.HandlePath(w, path)
+	}
+}
+
+func ok(w http.ResponseWriter) {
+	w.WriteHeader(http.StatusOK)
+	_, err := w.Write([]byte("ok"))
+	if err != nil {
+		panic(
+			fmt.Errorf(
+				"an unexpected error occurred during writing back the response in the context of the mockWebhookHandler",
+			),
+		)
+	}
+}
+
+func (rctrl RegistryCtrl) HandlePath(w http.ResponseWriter, path string) {
+	file, err := os.ReadFile(fmt.Sprintf("%s%s.json", rctrl.base, path))
```

Comment:
nit: I think we have the tooling to easily make this non-arb read
```suggestion
	fileName, err := utils.SafeFileName(rctrl.base, strings.Split(path, "/")...)
	if err != nil {
		w.WriteHeader(http.StatusNotFound)
		return
	}
	file, err := os.ReadFile(fileName)
```